sqlx-postgres/src/options/mod.rs

```diff
@@ -59,10 +60,9 @@ impl PgConnectOptions {
             .and_then(|v| v.parse().ok())
             .unwrap_or(5432);
 
-        let host = var("PGHOSTADDR")
-            .ok()
-            .or_else(|| var("PGHOST").ok())
-            .unwrap_or_else(|| default_host(port));
+        let host = var("PGHOST").ok().unwrap_or_else(|| default_host(port));
+
+        let host_addr = var("PGHOSTADDR").ok();
```

Comment:
Now not all strings are valid here, only numeric IPs are. But we are not doing *any* validation on the strings at all for all the sibling fields, so I did not start to do that here also. In fact there is not even an error path in this API. 

That I do not agree to the defaulting to ENVs in the first place was discussed in previous PRs, for the moment I consider the bugfix more important than improving tghe API so I just keep it "as wrong in my eyes as the siblings" while fixing host != hostaddr.